### PR TITLE
Am 495 allow menu items to be built from cms

### PR DESCRIPTION
**Pull request by sscaife**

### CODE REVIEW

File: src/SFA.DAS.Campaign.Api.UnitTests/Controllers/Menu/WhenGettingTheMenu.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                .ReturnsAsync(mediatorResult);
+
+            var controllerResult = await controller.GetMenuAsync(CancellationToken.None) as OkObjectResult;
+
+            var actualResult = controllerResult.Value as GetMenuResponse;
+            Assert.IsNotNull(actualResult);
+            actualResult.Menu.MainContent.Apprentices.Should().NotBeNullOrEmpty();
+            actualResult.Menu.MainContent.Employers.Should().NotBeNullOrEmpty();
+            actualResult.Menu.MainContent.Influencers.Should().NotBeNullOrEmpty();
+            actualResult.Menu.MainContent.TopLevel.Should().NotBeNullOrEmpty();
```
Comment (dashton82) [inline src/SFA.DAS.Campaign.Api.UnitTests/Controllers/Menu/WhenGettingTheMenu.cs:36]: Might be worth expanding on this with :
```
mockMediator.Verify(x=>x.Send(It.Is<GetMenuQuery>(c=>c.MenuType.Equals("TopLevel")), CancellationToken.None), Times.Once);
            mockMediator.Verify(x=>x.Send(It.Is<GetMenuQuery>(c=>c.MenuType.Equals("Apprentices")), CancellationToken.None), Times.Once);
            mockMediator.Verify(x=>x.Send(It.Is<GetMenuQuery>(c=>c.MenuType.Equals("Employers")), CancellationToken.None), Times.Once);
            mockMediator.Verify(x=>x.Send(It.Is<GetMenuQuery>(c=>c.MenuType.Equals("Influencers")), CancellationToken.None), Times.Once);
```

File: src/SFA.DAS.Campaign/ExternalApi/Requests/GetMenuRequest.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.Campaign.ExternalApi.Requests
+{
+    public class GetMenuRequest : IGetApiRequest
```
Comment (dashton82) [inline src/SFA.DAS.Campaign/ExternalApi/Requests/GetMenuRequest.cs:5]: Think there is a test missing for this 🤔

**Review by dashton82 — CHANGES_REQUESTED** [review]

**Review by dashton82 — DISMISSED** [review]

**Review by dashton82 — APPROVED** [review]

**Comment by sony-kotha** [comment]
Test completed.

**Comment by sony-kotha** [comment]
Testing compled.

**Comment by sony-kotha** [comment]
Testing completed.

**Pull request merged**